### PR TITLE
MM-20169 - Fixed escape characters used in FormattedMessage

### PR DESCRIPTION
**Pull request by marianunez**
#### Summary

react-intl that was recently upgraded in master and in v3 they [changed there escape character](https://github.com/formatjs/react-intl/blob/master/docs/Upgrade-Guide.md#escape-character-has-been-changed-to-apostrophe-)to `'` which was causing issues in formatting messages that were using that character around formatted values.  

#### Ticket Link
Fixes [MM-20169](https://mattermost.atlassian.net/browse/MM-20169)

### CODE REVIEW

**Comment by marianunez** [comment]
@enahum An escape character was added to the `en.json` file. The other translation files should not be updated in this case either?

File: i18n/en.json [hunk 1]
```diff
@@ -654,9 +654,9 @@
   "admin.group_settings.group_details.add_team_or_channel": "Add Team or Channel",
   "admin.group_settings.group_details.group_profile.name": "Name:",
   "admin.group_settings.group_details.group_teams_and_channels_row.remove": "Remove",
-  "admin.group_settings.group_details.group_teams_and_channels_row.remove.confirm_body": "Removing this membership will prevent future users in this group from being added to the '{name}' {displayType}.",
+  "admin.group_settings.group_details.group_teams_and_channels_row.remove.confirm_body": "Removing this membership will prevent future users in this group from being added to the ''{name}'' {displayType}.",
```
Comment (enahum) [inline i18n/en.json]: where are the changes in the code for this one

File: i18n/en.json [hunk 2]
```diff
@@ -654,9 +654,9 @@
...
   "admin.group_settings.group_details.group_teams_and_channels_row.remove.confirm_button": "Yes, Remove",
-  "admin.group_settings.group_details.group_teams_and_channels_row.remove.confirm_header": "Remove Membership from the '{name}' {displayType}?",
+  "admin.group_settings.group_details.group_teams_and_channels_row.remove.confirm_header": "Remove Membership from the ''{name}'' {displayType}?",
```
Comment (enahum) [inline i18n/en.json]: and this one?

File: i18n/en.json (hunk 1 above)
Comment (marianunez) [reply i18n/en.json]: Updated

File: i18n/en.json (hunk 2 above)
Comment (marianunez) [reply i18n/en.json]: ✅

**Comment by marianunez** [comment]
Given that the react-intl upgrade seems to be reverted in this PR: https://github.com/mattermost/mattermost-webapp/pull/4255 not sure if this should be merged. Adding the `Do not Merge` for now

**Comment by saturninoabril** [comment]
@marianunez Are these changes still valid?  I'm seeing duplicate single quote per this change.

**Comment by marianunez** [comment]
> @marianunez Are these changes still valid? I'm seeing duplicate single quote per this change.

Given that the `react-intl` upgrade has been reverted in https://github.com/mattermost/mattermost-webapp/pull/4255 then it is no longer relevant for v5.18. 

I guess we can close this PR and reopen after the upgrade is done later? 

@enahum Would you know if there is a place where I can leave this noted for the future upgrade?

**Comment by enahum** [comment]
@marianunez probably in the ticket? not sure really

**Comment by marianunez** [comment]
Closing this PR and noted in the upgrade `react-intl` ticket [MM-17555](https://mattermost.atlassian.net/browse/MM-17555) for future reference to reopen when the actual upgrade is done.

**Pull request closed without merging**